cyblog2.md

```diff
@@ -0,0 +1,70 @@
+1. create separate Cloud Functions for each Serverless callback.  One for listening and responding to the initial Slack events, and the second for processing the messages in the Topic.  However, here you can write a single Pulumi App where all the code can be placed how you like it (in this case in a single file).
+1. [package](https://cloud.google.com/functions/docs/writing/) each callback up in the appropriate structure Cloud Functions expects, including how to include or reference your dependencies properly.
+1. create a [Storage Bucket](https://cloud.google.com/storage/docs/creating-buckets) for all of our packaged programs to live in.
+1. upload each packaged program to a [Bucket Object](https://cloud.google.com/storage/docs/uploading-objects) in that bucket.
+1. configure the appropriate triggers on your Cloud Function stating how it should be triggered (for example, in response to an [HTTP trigger](https://cloud.google.com/functions/docs/calling/http) or a [Pub/Sub trigger](https://cloud.google.com/functions/docs/calling/pubsub)).  
+1. include the right information in the function so you can interact with your other cloud resources in the Pulumi App.  Without this, you would need to find a way to include that data in each Cloud Function's [environment variables](https://cloud.google.com/functions/docs/env-var) (or just hardcode them in '1') so that your program can access the rest of your cloud infrastructure.  In the above example, you can see how you can just reference your resources directly (like the PubSub Topic) *directly* from your Cloud Function callback.  Pulumi makes sure this all works, and that the data you use is available in that Cloud Function.
+1. figure out a safe way to encode and access secrets for your Cloud Function in case your code needs that to safely and securely accesss resources.  Here, we can just use Pulumi's [Config Secrets](https://pulumi.io/reference/config.html#secrets) to safely encrypt and make secrets available to your Cloud Function code.
+
+This is a lot to figure out and continually manage over the lifetime of your cloud application.  If you want to tweak things even slightly you might need to go make many manual changes and updates to ensure everything is properly updated.  However, in that single Pulumi App this is all taken care of for you.  For example, say you were to change the above example to use [Cloud Tasks](https://cloud.google.com/tasks/) insead of [PubSub](https://cloud.google.com/pubsub/).  You would just have to change to use the resurce you wanted near the top of the file.  You'd then get automatically told by TypeScript that you were using the resource improperly in your code.  Once you fixed it up, you could simply run `pulumi up`, and have the entirety of your stack properly updated (including having all those steps above automatically taken care of for you).
```

Comment:
```suggestion
This is a lot to figure out and continually manage over the lifetime of your cloud application.  If you want to tweak things even slightly you might need to go make many manual changes and updates to ensure everything is properly updated.  With Pulumi, all this complexity is handled with a single update!  For this example, changing the application to use [Cloud Tasks](https://cloud.google.com/tasks/) instead of [PubSub](https://cloud.google.com/pubsub/) only requires changing the initial resource declaration, and updating the references later in the code. Since we're using TypeScript, all of your normal refactoring tools are available and make it clear what needs to be updated. Once you're done refactoring, a simple `pulumi up` reconciles the changes and takes care of all the complexities we discussed above!
```